packages/synapse-react-client/src/components/TanStackTable/StyledTanStackTable.tsx

```diff
@@ -73,10 +73,15 @@ export default function StyledTanStackTable<
         header.column.getSize()
     }
     return colSizes
+    // Intentionally limit the dependencies to only recompute when the column sizes change
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [table.getState().columnSizingInfo, table.getState().columnSizing])
+  }, [
+    table.getState().columnSizingInfo,
+    table.getState().columnSizing,
+    table.getState().columnVisibility, // If a column is added, its width should be recalculated
```

Comment:
Fixes an occasional bug where these CSS variables would be initialized before SynapseTable had configured its visible columns.